### PR TITLE
Improve and update Keyring docs

### LINKED ISSUE #1030: Update Keyring API docs
Integrate latest updates from the Keyring API repository.

See:
- https://github.com/MetaMask/keyring-api/blob/main/README.md
- https://github.com/MetaMask/keyring-api/tree/main/docs

### PR DESCRIPTION
**Pull request by alexandratran**
Update the Keyring API content for clarity and integrate updates from latest releases. Additional edits to related content.
Fixes #1030 

Previews:
- About the Keyring API: https://docs.metamask.io/1030-keyring/snaps/concepts/keyring-api/
- How to create an account management Snap: https://docs.metamask.io/1030-keyring/snaps/how-to/use-keyring-api/create-account-snap/
- Account management Snap security guidelines: https://docs.metamask.io/1030-keyring/snaps/how-to/use-keyring-api/security/
- Create an account management companion dapp: https://docs.metamask.io/1030-keyring/snaps/how-to/use-keyring-api/create-companion-dapp/

### CODE REVIEW

File: snaps/concepts/keyring-api.md [hunk 1]
```diff
@@ -7,78 +7,65 @@ sidebar_custom_props:
...
+<img src={require('../assets/keyring/accounts-ui.png').default} alt="Account management Snap accounts in Metamask UI" width="360" style={{border: '1px solid gray'}} />
 </p>
 
-[Create a Keyring Snap to integrate custom EVM accounts in MetaMask.](../tutorials/custom-evm-accounts.md)
-Your dapp can then use the [`eth_requestAccounts`](/wallet/reference/eth_requestaccounts) MetaMask
-JSON-RPC API method to connect to the custom accounts, and seamlessly interact with them using other
-[JSON-RPC methods](/wallet/reference/json-rpc-api).
+To use the Keyring API, you first [implement the API in an account management Snap](../how-to/use-keyring-api/snap/index.md)
+(also known as an "account Snap").
+Your dapp can then [invoke Keyring API methods](../how-to/use-keyring-api/dapp.md) on the account
```
Comment (danroc) [inline snaps/concepts/keyring-api.md:20]: We've been calling the dapp that iteracts directly with the Snap a "companion dapp" to differentiate from dapps that iteract with the account through MetaMask.

File: snaps/how-to/use-keyring-api/dapp.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+---
+description: Call Keyring API methods from your dapp.
+sidebar_position: 2
+---
+
+# Use the Keyring API from a dapp
```
Comment (danroc) [inline snaps/how-to/use-keyring-api/dapp.md:6]: Should we use "companion dapp" here?

**Review by danroc — APPROVED** [review]

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
+---
+description: Get started by creating an account management Snap.
+sidebar_position: 1
+---
+
+# Create an account management Snap
+
+Create an account management Snap to connect to custom EVM accounts.
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:8]: The topic title and intro statement are almost the same. (In the Intelligent Content realm, the intro statement c/b the title.) Maybe the intro statement c/b... "The account management Snap connects to custom EVM accounts."

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+:::tip see also
+- [Create an account management companion dapp](create-companion-dapp.md)
+- [Account management Snap security guidelines](security.md)
+- [About the Keyring API](../../concepts/keyring-api.md)
+- [Keyring API reference](../../reference/keyring-api/index.md)
+:::
+
+## Prerequisites
+
+- Set up a Snap.
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:22]: Nice use of a prerequisite.

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+```typescript
+class MySnapKeyring implements Keyring {
+  // Implement the required methods here...
+}
+```
+
+### 4. Handle requests submitted by MetaMask
+
+MetaMask will submit Ethereum sign requests from dapps using the
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:76]: "MetaMask will submit..." = "MetaMask submits..." ?

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+
+Implement the [required Keyring API methods](security.md#limit-the-methods-exposed-to-dapps) in your Snap:
+
+```typescript
+class MySnapKeyring implements Keyring {
+  // Implement the required methods here...
+}
+```
+
+### 4. Handle requests submitted by MetaMask
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:74]: Handle = Manage ?

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+The request includes:
+
+- `id` - The unique identifier for the request.
+- `scope` - The CAIP-2 chain ID of the selected chain.
+  Currently, this property is always an empty string.
+  Your Snap should use the chain ID present in the request object instead.
+- `account` - The ID of the account that should handle the request.
+- `request` - The request object.
+
+Your Snap must respond with either a synchronous result:
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:107]: Is "either" necessary?

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+return { pending: false, result };
+```
+
+Or an asynchronous result:
+
+```typescript
+return { pending: true, redirect: { message, url } };
+```
+
+The redirect message and URL is displayed to the user to inform them on how to continue the transaction flow.
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:119]: Maybe... "The system displays the redirect message and URL to help the user continue the transaction flow."

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+### 5. Notify MetaMask about events
+
+Notify MetaMask when the following events take place, using the `emitSnapKeyringEvent()` helper function.
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:123]: What following events? Also, there's some duplication in the title and intro statement.

File: snaps/how-to/use-keyring-api/create-account-snap.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+
+1. An account is created:
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/create-account-snap.md:125]: The topic is step #5. Do you want to make the steps within this topic alphabetic...a, b, c...? Also, it seems that the numbered items here are results and not steps. Do you want to use bullets instead?

File: snaps/how-to/use-keyring-api/security.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+## Remove all debug code from your production Snap
+
+Ensure that all debug code is removed from your production Snap.
+Exposing debug code can lead to multiple security vulnerabilities.
+For example, secret information can be logged to the console, or a malicious dapp can bypass a
+security check.
+
+## Remove sensitive information from errors
+
+Ensure that all error messages returned by your Snap are sanitized.
```
Comment (mjsmike62) [inline snaps/how-to/use-keyring-api/security.md:154]: This sounds like a Caution or Danger admonition.

File: snaps/reference/rpc-api.md [hunk 12]
```diff
@@ -601,7 +601,7 @@ await snap.request({
 :::flaskOnly
 :::
 
-Manages [Keyring Snap](../concepts/keyring-api.md) accounts.
+Manages [account management Snap](../concepts/keyring-api.md) accounts.
```
Comment (mjsmike62) [inline snaps/reference/rpc-api.md:604]: This statement is a little difficult to read with the repetition of like words.

**Review by mjsmike62 — APPROVED** [review]
I have a few suggested edits.

**Pull request merged**